Выполнено ФЗ 12 спринта

a-dbr opened the pull request.

a-dbr committed:
Выполнено ФЗ 12 спринта

a-dbr committed:
код-стайл

a-dbr committed:
добавил пробел

Alexander-Bolotov commented on src/main/java/ru/yandex/practicum/filmorate/service/interfaces/GenreService.java:
```diff
@@ -0,0 +1,11 @@
+package ru.yandex.practicum.filmorate.service.interfaces;
```
(minor) обычно не принто называть пекедж по типу классов. Можно просто интерфейсы расположить в папке service, а реализацию в impl

Alexander-Bolotov commented on src/test/java/ru/yandex/practicum/filmorate/FilmServiceTest.java:
```diff
@@ -10,8 +10,9 @@
+import ru.yandex.practicum.filmorate.repository.interfaces.MpaRepository;
+import ru.yandex.practicum.filmorate.repository.interfaces.UserRepository;
 import ru.yandex.practicum.filmorate.service.interfaces.FilmService;
 
```
в тз просят сконфигурировать тесты через 
@JdbcTest
@AutoConfigureTestDatabase

Alexander-Bolotov commented on src/test/resources/application.properties:
```diff
@@ -0,0 +1,5 @@
+spring.datasource.url=jdbc:h2:mem:testdb;DB_CLOSE_DELAY=-1
```
jdbc:h2:file

Alexander-Bolotov commented on src/main/java/ru/yandex/practicum/filmorate/repository/mappers/FilmMapper.java:
```diff
@@ -1,26 +1,30 @@
-package ru.yandex.practicum.filmorate.repository;
+package ru.yandex.practicum.filmorate.repository.mappers;
```
маппере лучше вынести на один уровень выше по пекеджам

a-dbr committed:
Изменил структуру проекта.
Заменил ручные настройки бд для тестов на @AutoConfigureTestDatabase.
Убрал лишние аннотации.
